### ISSUE TITLE
Typescript error "Could not find a declaration file" when using NodeNext module resolution

### ISSUE THREAD

**Issue by christophemarois** [marpple/FxTS#187]
# Bug Report

### 💻 Code

When importing FxTS in a project where `compilerOptions.moduleResolution` is set to `NodeNext` in `tsconfig.json`, the typescript compiler fails to find the type definitions.

```ts
import { pipe } from '@fxts/core'
pipe /* any */
```

This can be easily fixed by setting `exports[*].types` to `./dist/types/index.d.ts` in FxTS's `package.json`:

```diff
{
  "exports": {
    "./package.json": "./package.json",
    ".": {
+     "types": "./dist/types/index.d.ts"
      "import": "./dist/esm/index.js",
      "require": "./dist/cjs/index.js"
   }
  },
}
```

Maybe all other exports listed could benefit from having this key added? AFAIK, this has no negative side-effect.

Related issue from another lib: https://github.com/microsoft/TypeScript/issues/51406

### 🙁 Actual behavior

Typescript doesn't find the definitions

### 🙂 Expected behavior

Typescript finds the definitions

---

### Version Information

- Operating system: macOS 13.1
- Typescript: 4.9.4
- Nodejs: 18.12.1

**Comment by christophemarois** [comment]
Another example of this can be found in [zod](https://github.com/colinhacks/zod/)'s `package.json`, where both `$.types` and `$.exports[*].types` are set to their `.d.ts` file

https://github.com/colinhacks/zod/blob/e6939195fbb5191402ba3d6f5b7aade463de6e51/package.json#L5-L17

**Comment by shine1594** [comment]
This issue has been fixed in version `0.12.1`. Thanks for reporting.
